fsspec/implementations/dbfs.py

```diff
@@ -1,66 +1,115 @@
-                   for o in r.json()["files"]]
+            try:
+                r = self.get_from_api("list", json={"path": path})
+            except requests.exceptions.HTTPError as e:
+                status_code = e.response.status_code
+                if status_code == 404:
+                    raise FileNotFoundError(path)
+                raise e
+            files = r["files"]
+            out = [
+                {
+                    "name": o["path"],
+                    "type": "directory" if o["is_dir"] else "file",
+                    "size": o["file_size"],
+                }
+                for o in files
+            ]
             self.dircache[path] = out
+
         if detail:
             return out
         return [o["name"] for o in out]
 
     def mkdirs(self, path, exist_ok=True):
         if not exist_ok:
             raise NotImplementedError
-        r = self.session.post(self.url + "/2.0/dbfs/mkdirs", json={"path": path})
-        r.raise_for_status()
+        self.post_to_api("mkdirs", json={"path": path})
 
-    def makedir(self, path, create_parents=True, **kwargs):
+    def makedirs(self, path, create_parents=True, **kwargs):
         if not create_parents:
             raise NotImplementedError
-        self.mkdirs(path)
+        self.mkdirs(path, **kwargs)
 
     def rm(self, path, recursive=False):
-        r = self.session.post(self.url + "/2.0/dbfs/delete",
-                              json={"path": path, "recursive": recursive})
-        r.raise_for_status()
+        self.post_to_api("delete", json={"path": path, "recursive": recursive})
 
-    def mv(self, path1, path2, recursive=False, maxdepth=None, **kwargs):
+    def mv(self, path1, path2, recursive=False, maxdepth=None):
         if recursive:
             raise NotImplementedError
-        r = self.session.post(self.url + "/2.0/dbfs/move",
-                              json={"source_path": path1, "destination_path": path2})
+        if maxdepth:
+            raise NotImplementedError
+
+        self.post_to_api(
+            "move",
+            json={"source_path": path1, "destination_path": path2},
+        )
+
+    def create_handle(self, path, overwrite=True):
+        r = self.post_to_api("create", json={"path": path, "overwrite": overwrite})
+        return r["handle"]
+
+    def close_handle(self, handle):
+        self.post_to_api("close", json={"handle": handle})
+
+    def add_data(self, handle, data):
+        data = base64.b64encode(data).decode()
+        self.post_to_api(
+            "add-block",
+            json={"handle": handle, "data": data},
+        )
+
+    def get_data(self, path, start, end):
+        r = self.get_from_api(
+            "read",
+            json={"path": path, "offset": start, "length": end - start},
+        )
+        return base64.b64decode(r["data"])
+
+    def post_to_api(self, endpoint, json):
+        r = self.session.post(os.path.join(self.url, "2.0/dbfs", endpoint), json=json)
```

Comment:
os.path.join will do the wrong thing on windows. urlparse has a join, but I think "/" or just `+` are fine.